### ISSUE TITLE
What's the rule for version numbers?

### ISSUE THREAD

**Issue by SubaruArai** [raspberrypi/pico-sdk#902]
## TLDR

SDK1.4.0 has some breaking changes, which seems to contradict semantic versioning.
Can someone tell me what the versioning rules are?

## long version

The raspberry pi team had done an excellent job for the SDK1.4.0, and I'm really excited about it!
But before trying it out, I found some problems in the release note.

### The problem

In the miscellaneous section of pico-sdk 1.4.0 release note, there was this line:

> - The minimum PLL frequency has been increased from 400Mhz to 750Mhz to improve stability across operating conditions. This should not affect the majority of users in any way, but may impact those trying to set particularly low clock frequencies. If you do wish to return to the previous minimum, you can set `PICO_PLL_VCO_MIN_FREQ_MHZ` back to `400`. There is also a new `PICO_PLL_VCO_MAX_FREQ_MHZ` which defaults to `1600`.

Although minor, this seems like a breaking change. i.e. A project using SDK1.3.1 might fail to build or run on SDK1.4.0.
While I'm not affected by this change, someone might by assuming that pico-sdk is based on semantic versioning.

### The question/request

1. What's the rule for pico-sdk's versioning?
2. Is it is semantic versioning, I highly recommend changing `1.4.0` to `2.0.0`, and add a `1.4.1` that reverts the breaking changes.
3. The first line in the release note's miscellaneous section, it stated a function has moved from one file to another. It should be noted that this isn't a breaking change, since `hardware/sync.h` includes `pico/platform.h`.

**Comment by SubaruArai** [comment]
@aallan 
I'm fine with closing so long as there is a proper reason, can you explain to me?
